user-guide/Advanced_Functionality/DataMiner_Systems/BrokerGateway_Migration.md

```diff
@@ -0,0 +1,149 @@
+
+A successful installation output might look like this:
+
+```cmd
+C:\Skyline DataMiner\Tools>NATSMigration.exe
+Warning: This will alter DataMiner NATS configuration.
+This will also restart DataMiner. Continue? (y/n):y
+LAURENSVG.skyline.local - Stopping DataMiner.
+************  Stopping the DataMiner software   *************
+
+...
+
+(un)install BrokerGateway managed NATS server? (install/uninstall):install
+LAURENSVG.skyline.local - BrokerGateway SoftLaunch flag is set to True.
+LAURENSVG.skyline.local - Setting the MessageBrokerConfig.json...
+SLKill called for "nats" with force=False
+*** Stop services ***
+Service (NATS) has been stopped.
+*** Kill processes ***
+Process (nats-account-server : 55500) has been killed.
+SLKill called for "nas" with force=False
+*** Stop services ***
+Service (NAS) has been stopped.
+*** Kill processes ***
+LAURENSVG.skyline.local - DataMiner BrokerGateway is already running.
+LAURENSVG.skyline.local - DataMiner BrokerGateway is reachable.
+LAURENSVG.skyline.local - Agent that will set up the cluster: 10.3.1.54
+LAURENSVG.skyline.local - This agent is resetting the cluster...
+LAURENSVG.skyline.local - checking if BGW of 10.3.1.54 is running...
+LAURENSVG.skyline.local - BGW of 10.3.1.54 is running.
+LAURENSVG.skyline.local - Calling ResetCluster endpoint.
+LAURENSVG.skyline.local - Contacted nats-server and BrokerGateway successfully!
+LAURENSVG.skyline.local - Starting DataMiner.
+*************************************************************
+***********  Starting the DataMiner software   ************
+*************************************************************
+--
+-- This tool restarts the DataMiner software, including SLNet
+--
+--
+-- Starting SLNet
+The SLNet Service service is starting.
+The SLNet Service service was started successfully.
+
+-- Starting DataMiner
+************  Starting the DataMiner software   *************
+
+The SLDataMiner service is starting.
+The SLDataMiner service was started successfully.
+
+--
+-- The DataMiner software is starting
+--
+*************************************************************
+(c) Skyline Communications                     www.skyline.be
+```
+
+## What does migration do?
+
+First, it will stop DataMiner.
+
+Secondly, it will set the BrokerGateway softlaunch flag to true. This flag can be found in the **C:\Skyline DataMiner\SoftLaunchOptions.xml** file. Content might look like this:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<SLNet>
+    <BrokerGateway>true</BrokerGateway>
+</SLNet>
+```
+
+Afterwards, it will write the MessageBrokerConfig.json file at **C:\ProgramData\Skyline Communications\DataMiner\MessageBrokerConfig.json**. This file is used when creating default Sessions using the DataMinerMessageBroker.API(.Native) library. The file will be rewritten to reference the BrokerGateway url and api key path. Content might look similar to this:
```

Comment:
I just moved this in [Remove maximum nodecount for NATS clusters](https://devcore3/review/#/c/96526/).
It now is changed after Resetcluster has been called. (sorry!)